nfs/src/rpc_task.cpp

```diff
@@ -1313,30 +1273,49 @@ void rpc_task::run_write()
      * membufs. We don't wait for the writes to actually finish, which means
      * we support buffered writes.
      */
-    std::vector<bytes_chunk> bc_vec =
-            copy_to_cache(this, ino, bufv, offset, length, error_code);
+    uint64_t periodic_bytes = 0;
+
+    copy_to_cache(this, ino, bufv, offset, length, error_code, &extent_left, &extent_right);
     if (error_code != 0) {
         AZLogWarn("[{}] copy_to_cache failed with error={}, "
                   "failing write!", ino, error_code);
         reply_error(error_code);
         return;
     }
 
-    assert(bc_vec.size() > 0);
+    assert(extent_right >= extent_left);
 
-    for (bytes_chunk& bc : bc_vec) {
-        /*
-         * Create the flush task to carry out the write.
-         */
-        struct rpc_task *flush_task =
-            get_client()->get_rpc_task_helper()->alloc_rpc_task(FUSE_FLUSH);
-        flush_task->init_flush(nullptr /* fuse_req */, ino);
+    /*
+     * Get the prune goals, if it's more than data already in-flight
+     * data to BLOB, then issue fresh flush.
+     */
+    inode->filecache_handle->get_prune_goals(nullptr, &periodic_bytes);
+    AZLogDebug("periodic_bytes: {}, extent_left{}, extent_right: {}",periodic_bytes, extent_left, extent_right);
+
+    if (inode->inflight_bytes >= periodic_bytes) {
+        AZLogDebug("Reply write without syncing to Blob");
+        reply_write(length);
+        return;
+    }
```

Comment:
this is not the right way to decide if we need to send data to server now.
Note that we are not sending data because dirty data has exceeded some threshold, but instead we would like to send as soon as we have large enough dirty data. As of now, large enough is anything >= wsize, but later it'll be the block size of the corresponding block in the blob. In any case as soon as we have large enough data it doesn't help to wait more, so we should send immediately irrespective of how the dirty data compares against the prune limits.